src/Microsoft.AspNet.Mvc.Core/Controllers/ServiceBasedControllerActivator.cs

```diff
@@ -13,19 +13,26 @@ namespace Microsoft.AspNet.Mvc.Controllers
     public class ServiceBasedControllerActivator : IControllerActivator
     {
         /// <inheritdoc />
-        public object Create(ActionContext actionContext, Type controllerType)
+        public object Create(ControllerContext actionContext)
         {
             if (actionContext == null)
             {
                 throw new ArgumentNullException(nameof(actionContext));
             }
 
-            if (controllerType == null)
-            {
-                throw new ArgumentNullException(nameof(controllerType));
-            }
+            var controllerType = actionContext.ActionDescriptor.ControllerTypeInfo?.AsType();
```

Comment:
Should this throw if the `ControllerTypeInfo` is null? `GetRequiredService` might throw, but it might not be a particularly useful message.